addons/web/static/src/views/fields/parsers.js

```diff
@@ -151,45 +196,62 @@ export function parsePercentage(value) {
     return parseFloat(value) / 100;
 }
 
+/**
+ * Extract the string to be parsed as float from [str] given the [currency].
+ * @param {string} str
+ * @param {Currency} currency
+ */
+function extractMonetaryValue(str, currency) {
+    const symbol = escapeRegExp(currency.symbol);
+    let re;
+    if (currency.position === "after") {
+        re = new RegExp(`^(?<thevalue>.*([^\\s]{1}))[\u00a0]*(${symbol})[\\s]*$`);
+    } else {
+        re = new RegExp(`^[\\s]*(${symbol})[\u00a0]*(?<thevalue>[^\\s]{1}.*)$`);
+    }
+    const match = str.match(re);
+    if (match === null) {
+        // if match is null, it means that the currency symbol is not present in the string
+        // or the currency symbol is present but not at the right position
+        return str;
+    }
+    return match.groups.thevalue.trim();
+}
+
 /**
  * Try to extract a monetary value from a string. The localization is considered in the process.
  * The monetary value can have the formats sym$&nbsp;float, float$&nbsp;sym or float
  * where $&nbsp; is a non breaking space and sym is a currency symbol.
  * If a symbol is found it must correspond to the default currency symbol or to the
  * symbol of the currency whose id is passed in options.
  *
+ * @typedef Currency
+ * @property {string} symbol
+ * @property {"before" | "after"} position
+ * @property {[69, 2]} digits
+ *
  * @param {string} value
  * @param {Object} [options={}]
  * @param {number} [options.currencyId]
  * @returns {number} float
  */
 export function parseMonetary(value, options = {}) {
-    // TODO GES help ?
-    // const values = value.split("&nbsp;");
-    const values = value.split(nbsp);
-    if (values.length === 1) {
-        return parseFloat(value);
-    }
-    let currency = session.currencies[options.currencyId];
-    if (!currency) {
-        if (Object.keys(session.currencies).length !== 0) {
-            // BS
-            currency = session.currencies[Object.keys(session.currencies)[0]];
-        } else {
-            throw new InvalidNumberError(
-                `"${value}" is either an invalid number or is using an unconfigured currency symbol`
-            );
-        }
-    }
-    const symbolIndex = values.findIndex((v) => v === currency.symbol);
-    if (symbolIndex === -1) {
-        throw new InvalidNumberError(`"${value}" doesn't have the expected currency symbol`);
+    /** @type {Currency | undefined} */
+    const currency = session.currencies[options.currencyId];
```

Comment:
can't we have a simple, maybe more naive logic, like
- removing the symbol (of the currency given in options or the default currency if no currency given), if the value starts/ends with (depending on the currency if we want to be strict)
- trimming the value
- calling parseNumber